android/src/main/java/com/raulh82vlc/MoviesFinder/ui/activities/MovieDetailsActivity.java

```diff
@@ -190,13 +191,20 @@ protected void setInitialTitle() {
 
     @Override
     protected void setToolbarInitialisation() {
-        mToolbar.setNavigationIcon(R.drawable.ic_back);
+        //mToolbar.setNavigationIcon(R.drawable.ic_back);
         setSupportActionBar(mToolbar);
-        final ActionBar actionBar = getSupportActionBar();
 
-        if (actionBar != null) {
-            actionBar.setDisplayHomeAsUpEnabled(true);
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            getSupportActionBar().setHomeButtonEnabled(true);
         }
+
+        mToolbar.setNavigationOnClickListener(new View.OnClickListener() {
```

Comment:
This `OnClickListener` is unnecessary here, since the back functionality is enabled by default at the toolbar when you set those previous flags to `true`, you can get rid of this bit of code.